game/src/environment/portal/Portal.gd

```diff
@@ -26,4 +29,9 @@ func _on_Base_body_entered(body: Node) -> void:
 		var enemy: Enemy = body
 		teleport(enemy)
 		emit_signal("take_damage", body.damage, health)
+		var dmg = body.damage * -1
```

Comment:
And this one better not keep health - dmg = new health logic in multiple places, if you want to implement future things it will be a pain. Stick to keeping portal health being the source of truth and sending in self.heath as input rather than damage as input. The cleanest way would be to rewrite the healthbar's function so that this works:

current:

```
	var dmg = body.damage * -1
		if (isRed):
			health_bar._change_red_portal_health(dmg)
		else:
			health_bar._change_blue_portal_health(dmg)
```

new:

```
    health_bar.change_portal_health(self.health, isRed)
```

Reply:
Also btw _change_red_portal starts with an underscore, which by convention means a private variable not called by any other class or object. since you intend to call it elsewhere, i suggest dropping the leading underscore

Reply:
fixed.